jina/types/document/__init__.py

```diff
@@ -295,16 +300,15 @@ def _update_doc(d: Dict):
                 # fortunately protobuf throws a warning when the parsing seems go wrong
                 # the context manager below converts this warning into exception and throw it
                 # properly
-                with warnings.catch_warnings():
-                    warnings.filterwarnings(
-                        'error', 'Unexpected end-group tag', category=RuntimeWarning
-                    )
-                    try:
-                        self._pb_body.ParseFromString(document)
-                    except RuntimeWarning as ex:
-                        raise BadDocType(
-                            f'fail to construct a document from {document}'
-                        ) from ex
+                filters = warnings.filters
```

Comment:
```suggestion
                filters = copy(warnings.filters)
```
I'm not sure whether the `copy` can be ignored here? Maybe you should **add a unit test** for this feature.

Reply:
@numb3r3 do you know cases where not using `copy` can break something ?
here, we're just switching the reference of warnings.filters which preserves the integrity of the filters object and performs faster than copying the underlying objects inside the list

Reply:
I am also not sure it makes so much sense this copy, u will be changing the same underlying object right?

Reply:
Can u add a tests that forces `ParseFromString` to emit some warnings and capture it properly